server/routes/ad.ts

```diff
@@ -460,13 +460,16 @@ const getDetectors = async (
         );
         return detectorStateResp;
       } catch (err) {
-        console.log(
-          'Anomaly detector - Unable to retrieve detector state',
-          err
+        return Promise.reject(
```

Comment:
I would prefer you log the `err`. Same for below changes.

Reply:
Sure, I can add in more error logs. There will still be an error log when the thrown errors get caught in the outermost `catch()` block.